lib/spack/spack/fetch_strategy.py

```diff
@@ -392,7 +392,26 @@ def __init__(self, *args, **kwargs):
 
     @_needs_stage
     def fetch(self):
-        super(CacheURLFetchStrategy, self).fetch()
+        path = re.sub('^file://', '', self.url)
+
+        # directly check whether the cache file already exists.
+        if not os.path.isfile(path):
+            raise NoCacheError('No cache of %s' % path)
+
+        # Just symlink it in if so.
+        self.stage.chdir()
+        basename = os.path.basename(path)
+        if os.path.exists(basename):
+            link = os.path.abspath(os.path.realpath(basename))
+            target = os.path.abspath(os.path.realpath(path))
+            if link != target:
```

Comment:
So basically if the link is already pointing to the right place, do nothing. Otherwise reset the link. Is there a possible error from removing the link unconditionally (and I presume it would also be fast to always regenerate it). Either way a comment would be helpful.